website/app/components/doc/page/header/search/index.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+import { autocomplete } from '@algolia/autocomplete-js';
+
+import { getItemsFunction } from './parts/getItemsFunction';
+import { templatesHeaderFunction } from './parts/templatesHeaderFunction';
+// import { templatesFooterFunction } from './parts/templatesFooterFunction';
+import { templatesItemFunction } from './parts/templatesItemFunction';
+import { htmlPanelFooter } from './parts/htmlPanelFooter';
+import { htmlPanelNoResults } from './parts/htmlPanelNoResults';
+
+export default class DocAlgoliaSearchComponent extends Component {
```

Comment:
Would we want to include the name of the tool used in the component name like this ("Algolia")?

Reply:
In this case yes, it's very specific and it would be a super minor thing to change it, compared with the effort to use another tool. It's to avoid confusion with a very generic term like "search" (that in some cases is used to mean "filter", hence https://github.com/hashicorp/design-system/pull/1818/files)